format/index.md

```diff
@@ -113,9 +113,14 @@ of resources, or resource relationship primarily targeted by a request.
 
 A document **MUST** contain either primary data or an array of [error objects](#errors).
 
-Primary data **MUST** appear under a top-level key named `"data"`. Primary
-data **MUST** be either a single resource object, an array of resource
-objects, or a value representing a resource relationship.
+Primary data **MUST** appear under a top-level key named `"data"`.
+
+Primary data **MUST** be either:
+
+* a single resource object or `null`, for requests that target single resources
+* an array of resource objects or an empty array ([]), for requests that target
+resource collections
+* resource linkage, for requests that target a resource's relationship
```

Comment:
Sorry, I totally missed the `null` case. I had actually sketched the exact same list as you did here (sans the nulls) before settling on the separate paragraph I quoted. You're right that the second sentence seems kinda redundant now, but maybe it could remain for completeness. I don't really have an opinion either way.